### DISCUSSION TITLE
Can ingest.py be modified to ingest code, like Python, Go, C, etc? I want to ask questions of my codebase

### DISCUSSION THREAD

**Discussion by zorrobyte** [imartinez/privateGPT#572] [Q&A]
Greetings!

I have a complex open source project that I contribute to, it would be nice to ingest the GitHub repo along with the wiki of the project to ask code and other questions of the project.

This seems to be as easy as adding a mapping for .py, .go, .h files, etc in https://github.com/imartinez/privateGPT/blob/main/ingest.py#L65

Actually, just did some research and it looks like: https://python.langchain.com/en/latest/modules/indexes/document_loaders/examples/git.html

Holy cow, so I can just ingest just about whatever I want and query it, eh?

Of course, I'm sure there's something I'm missing here. I'm very open to ideas :)

**Comment by johnbrisbin** [comment]
It may be just as easy as you suggest, assuming the loader in GitLoader work just like the others. As noted in that document, you need to import the module, then add the loader and file suffix to the table in ingest.py. That table replaces the method used in the Filtering files to load section you referred to. Give it a try. Make a contribution to the project.
**"But then, nothing is ever so simple as it first appears."**

**Comment by rollsch** [comment]
I am keen to try this. How did you go with trying it?

**Comment by rollsch** [comment]
Seems this loader is more specific to git repos. Do you know what underlying loader it uses to parse code itself? Id like to parse individual C files.

I tried adding it to ingest and adding a path to a git repo, though it will need some more changes as it treats my repo as an individual document currently (which obviously wont work)

Though I can probably hack ingest.py up to make a fake git repo with all my C files in it. I will report back.

VERY INTERESTING STUFF!

**Comment by Pursche** [comment]
I am also very interested in using this to ingest a C++ project, I would love to hear back if you figure it out!
